OpenRA.Mods.Common/Widgets/WorldCommandWidget.cs

```diff
@@ -191,15 +191,28 @@ bool CycleBases()
 				.Select(b => b.Actor)
 				.ToList();
 
-			if (!bases.Any())
-				return true;
+            var noBase = false;
+
+            // If no base found the nondestroyed buildings are selected.
+            if (!bases.Any())
+            {
+                bases = world.ActorsWithTrait<TargetableBuilding>()
+                .Where(a => a.Actor.Owner == world.LocalPlayer)
+                .Select(b => b.Actor)
+                .ToList();
+                noBase = true;
+            }
+
+            if (!bases.Any())
```

Comment:
You can move this if inside the other if so you don't check it twice if its false in both cases.